openlibrary/plugins/openlibrary/js/ol.js

```diff
@@ -390,186 +390,251 @@ function passwordHide(){
+
+    var marshalBookSearchQuery = function(q) {
+        if (q.indexOf(':') == -1 && q.indexOf('"') == -1) {
+           q = 'title: "' + q + '"';
+        }
+        return q;
+    }
+
+    var renderSearchResults = function(q) {
+        var facet_value = searchFacets[localStorage.getItem("facet")];
+        if (q === '') {
+            return;
+        }
+        if (facet_value === 'books') {
+            q = marshalBookSearchQuery(q);
+        }
+
+        var url = composeSearchUrl(q, true, 10);
+
+        $('header .search-component ul.search-results').empty()
+        $.getJSON(url, function(data) {
+            for (var d in data.docs) {
+                renderSearchResult[facet_value](data.docs[d]);
+            }
+        });
     }
 
     var setFacet = function(facet) {
-	localStorage.setItem("facet", facet.toLowerCase());
-	$('header .search-facet-selector select').val(capitalize(localStorage.getItem("facet")))
-	$('header .search-facet-value').html(capitalize(localStorage.getItem("facet")));
-    	var url = ((localStorage.getItem("facet") === 'books')? '/search' : "/search/" + localStorage.getItem("facet"));
-	console.log(url);
-	$('.search-bar-input').attr("action", url);
+        var facet_key = facet.toLowerCase();
+
+        if (facet_key === 'advanced') {
+            localStorage.setItem("facet", '');
+            window.location.assign('/advancedsearch')
+            return;
+        }
+
+        localStorage.setItem("facet", facet_key);
+        $('header .search-facet-selector select').val(capitalize(facet_key))
+        $('header .search-facet-value').html(capitalize(facet_key));
+        $('header .search-component ul.search-results').empty()
+        q = $('header .search-component .search-bar-input input').val();
+        var url = composeSearchUrl(q)
+        $('.search-bar-input').attr("action", url);
+        renderSearchResults(q);
     }
     var options = Browser.getJsonFromUrl();
-    setFacet(options.facet || localStorage.getItem("facet") || "books");
-
-    if (options.q) {	
-	$('.search-bar-input [type=text]').val(options.q.replace(/\+/g, " "));
+    setFacet(options.facet || localStorage.getItem("facet") || "title");
+
+    if (options.q) {
+        var q = options.q.replace(/\+/g, " ")
+        if (q.indexOf('title:') != -1) {
+            var parts = q.split('"');
+            if (parts.length === 3) {
+                q = parts[1];
+            }
+        }
+        $('.search-bar-input [type=text]').val(q);
     }
 
     var debounce = function (func, threshold, execAsap) {
-	var timeout;
-	return function debounced () {
-	    var obj = this, args = arguments;
-	    function delayed () {
-		if (!execAsap)
-		    func.apply(obj, args);
-		timeout = null;
-	    };
-	    
-	    if (timeout) {
-		clearTimeout(timeout);
-	    } else if (execAsap) {
-		func.apply(obj, args);
-	    }
-	    timeout = setTimeout(delayed, threshold || 100);
-	};
+        var timeout;
+        return function debounced () {
```

Comment:
Remove space between function name and `()`. Heck, shouldn't we be using a JS linter so we don't have to think about things like that?